### PR TITLE
Remove HLRC docs and related integration tests.

### LINKED ISSUE #81299: Remove HLRC docs in 8.0 and master branches
From 8.0 and forward, the HLRC client will no longer be published. 
It currently exists in the codebase (in master and 8.0 branches), 
because a significant number of tests use it. So effectively it is test code now.

The docs for HLRC still exist in the master and 8.0 branches. I think we should remove these docs?
Or at least replace it with a redirect section about the new high level rest client?
Otherwise this can cause confusion about the old HLRC existence?

### PR DESCRIPTION
**Pull request by martijnvg**
The HLRC will no longer be published from 8.0.0 and onwards.
Also the HLRC docs are no longer published, so these can be removed now.

The HLRC is currently used in a number of java rest tests and
for this reason it can't be removed completely. SO the HLRC is 
currently just an internal test rest client.

Closes #81299

### CODE REVIEW

File: docs/java-rest/high-level/index.asciidoc [hunk 1]
```diff
@@ -6,33 +6,11 @@
 [partintro]
 --
 
-deprecated[7.15.0, The High Level REST Client is deprecated in favour of the {java-api-client}/index.html[Java API Client].]
+removed[8.0.0, The High Level REST Client has been removed in favour of the {java-api-client}/index.html[Java API Client].]
```
Comment (martijnvg) [inline docs/java-rest/high-level/index.asciidoc:9]: This page is the only non remove diff of this PR.
Should we add more information here? Or should this page be completely removed as well?

**Comment by martijnvg** [comment]
@debadair I see that the docs for low level client are also no longer built (#78602), so I think we delete the entire [java-rest](https://github.com/elastic/elasticsearch/tree/master/docs/java-rest) directory?

**Comment by debadair** [comment]
> @debadair I see that the docs for low level client are also no longer built (#78602), so I think we delete the entire [java-rest](https://github.com/elastic/elasticsearch/tree/master/docs/java-rest) directory?

The low level docs are now part of the Java client docs, but they're being pulled from this directory. It would be cleaner to move them & update the includes so we *can* remove the entire directory.

**Comment by martijnvg** [comment]
> The low level docs are now part of the Java client docs, but they're being pulled from this directory. It would be cleaner to move them & update the includes so we can remove the entire directory.

Agreed. I think as a first step the hlrc docs can be removed? Then the LLRC docs can be moved to [the docs section of the java ES client](https://github.com/elastic/elasticsearch-java/tree/main/docs)?

**Review by dakrone — APPROVED** [review]
LGTM, I think this will resolve https://github.com/elastic/elasticsearch/issues/73317 and https://github.com/elastic/elasticsearch/issues/71210 also (since those files have now been removed)

**Review by debadair — APPROVED** [review]

**Pull request merged**